perfkitbenchmarker/disk.py

```diff
@@ -172,6 +172,8 @@ class BaseDiskSpec(spec.BaseSpec):
     mount_point: None or string. Directory of mount point.
     num_striped_disks: int. The number of disks to stripe together. If this is
         1, it means no striping will occur. This must be >= 1.
+    os_type: None or string. OS type of the VM to which the disk will be
```

Comment:
Attach gets complicated with all the per-provider overriding. It may make sense to just remove the FLAGS.os_type checks entirely. It seems like the use cases are pretty well partitioned by how we've split the code into linux_benchmarks, linux_packages, and linux_virtual_machine vs windows_benchmarks, windows_packages, and windows_virtual_machine. Also, at this point most providers override GetDevicePath and ignore this check anyway. I can't even find the use case for GetDeviceId. @noahl opened #741, so perhaps he can comment on this.